src/qtils/strict_sptr.hpp

```diff
@@ -44,110 +43,174 @@ namespace qtils {
       }
     }
 
-   private:
-    struct ForceInject {};  // Auxiliary arg appended to select ctor for DI
+    /**
+     * @brief Constructs from raw pointer and custom deleter
+     * @param raw Raw pointer to managed object
+     * @param deleter Deleter to be used
+     * @throws std::invalid_argument if raw is null
+     */
+    StrictSharedPtr(T *raw, Deleter deleter)
+        : ptr_(std::shared_ptr<T>(raw, std::move(deleter))) {
+      if (!ptr_) {
+        throw std::invalid_argument("Shared pointer is null");
+      }
+    }
 
-   public:
-    StrictSharedPtr(std::shared_ptr<T> sptr, ForceInject)
+  private:
+    /**
+     * @brief Special tag type to select the DI constructor overload.
+     *
+     * This struct is used to differentiate the constructor intended
+     * specifically for Dependency Injection (DI) frameworks.
+     */
+    struct ForceInject {};
+
+  public:
+    /**
+     * @brief Constructor for use by Dependency Injection (DI) frameworks (e.g., Boost.DI).
+     *
+     * DI frameworks typically prefer constructors with the maximum number
+     * of parameters when resolving dependencies. This overload includes
+     * an extra dummy tag (`ForceInject`) and a variadic parameter pack (`...`) to
+     * ensure it has the highest arity and is therefore selected by the injector.
+     *
+     * Unlike the standard constructor, this version **bypasses the null-check**.
+     * It assumes the DI container guarantees that the passed `shared_ptr`
+     * is valid and correctly constructed.
+     *
+     * The constructor is not intended for general use. It can only be called
+     * explicitly with the `ForceInject` tag to avoid accidental misuse.
+     *
+     * @param sptr The shared_ptr to wrap.
+     * @param Dummy The ForceInject tag to select this constructor.
+     * @param ... Additional dummy parameters to increase constructor arity.
+     */
+    StrictSharedPtr(std::shared_ptr<T> sptr, ForceInject, ...)
```

Comment:
Kagome used `Foo(Inject, ...);` instead of `Foo(Inject, ...) = delete;` to allow injector choose (sfinae/concept) this longest constructor, but fail at link time.
Using `(...)` in successful constructor looks suspicious, what kind of values will injector pass to this C varargs at runtime?
```suggestion
    StrictSharedPtr(std::shared_ptr<T> sptr, ForceInject, ForceInject)
```
or
```c++
template <typename T>
struct boost::di::ctor_traits<qtils::StrictSharedPtr<T>> {
  BOOST_DI_INJECT_TRAITS(std::shared_ptr<T>);
};
```